### PR TITLE
feat: Info Card

### PR DESCRIPTION
**Pull request by sooinc**
### Overview
Ports `InfoCard` component used for ProjectCards in portal-app.
We will be using this in the monolith (see [PR](https://github.com/codecademy-engineering/Codecademy/pull/28640) here).
And we have a tech debt ticket to replace the existing `InfoCard` component with this Gamut one (see [ticket](https://codecademy.atlassian.net/jira/software/projects/EGG/boards/142/backlog?selectedIssue=EGG-2083))

- Add InfoCard

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/Lk1KXQGDIrFhxvzs5jO0pr/Project-Hub-Lab?node-id=545%3A55688
- [x] Related to JIRA ticket: Egg-2082
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.32-alpha.48eada.0<br/>@codecademy/gamut-labs@28.4.1-alpha.48eada.0<br/>@codecademy/gamut@42.1.1-alpha.48eada.0<br/>@codecademy/monaco@0.5.24-alpha.48eada.0<br/>@codecademy/styleguide@52.2.1-alpha.48eada.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b8bfe380cf300060ee8322--gamut-preview.netlify.app>https://61b8bfe380cf300060ee8322--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b8bfe380cf300060ee8322>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.34-alpha.a25260.0<br/>@codecademy/gamut-labs@28.4.3-alpha.a25260.0<br/>@codecademy/gamut@42.1.2-alpha.a25260.0<br/>@codecademy/monaco@0.5.25-alpha.a25260.0<br/>@codecademy/styleguide@52.2.3-alpha.a25260.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b9161007198e3c7ff961a2--gamut-preview.netlify.app>https://61b9161007198e3c7ff961a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b9161007198e3c7ff961a2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.34-alpha.0f1707.0<br/>@codecademy/gamut-labs@28.4.3-alpha.0f1707.0<br/>@codecademy/gamut@42.1.2-alpha.0f1707.0<br/>@codecademy/monaco@0.5.25-alpha.0f1707.0<br/>@codecademy/styleguide@52.2.3-alpha.0f1707.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b9189cf7e6c048c616b9a5--gamut-preview.netlify.app>https://61b9189cf7e6c048c616b9a5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b9189cf7e6c048c616b9a5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.34-alpha.ef032b.0<br/>@codecademy/gamut-labs@28.4.3-alpha.ef032b.0<br/>@codecademy/gamut@42.1.2-alpha.ef032b.0<br/>@codecademy/monaco@0.5.25-alpha.ef032b.0<br/>@codecademy/styleguide@52.2.3-alpha.ef032b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b93440ada794008fb4b758--gamut-preview.netlify.app>https://61b93440ada794008fb4b758--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b93440ada794008fb4b758>Deploy Logs</a></p>

**Review by natalieweesh — APPROVED** [review]
LGTM

**Review by dreamwasp — APPROVED** [review]

File: packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const href = 'www.codecademy.com';
+const renderView = setupRtl(InfoCard, { topText, title, subtitle, body, href });
+
+const clickHandler = jest.fn();
+
+describe('InfoCard', () => {
+  it('onClick function is called when InfoCard is clicked', () => {
+    const { view } = renderView({ onClick: clickHandler });
+    view.getByText(topText).click();
+    expect(clickHandler).toHaveBeenCalled();
```
Comment (MattSchiller) [inline packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx]: nit: Can we change this to `.toHaveBeenCalledTimes(1)`? It happens infrequently, but using these targeted number of times have helped me find double-calls in the past.

File: packages/gamut/src/InfoCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  title: string;
+  subtitle: string;
+  body?: string;
+  bottomLeftText?: string;
+  bottomRightTagText?: string;
+  bottomRightTagColor?: TagColor;
+  cardHeight?: string | number;
+};
+
+export type TagColor = 'blue' | 'green' | 'pink';
```
Comment (MattSchiller) [inline packages/gamut/src/InfoCard/index.tsx]: a minor type thing, but I'd recommend the following:
```ts
const tagColorMap: Record<TagColor, string> = {
  blue: 'paleBlue',
  green: 'paleGreen',
  pink: 'palePink',
};
```

It keeps the type more specific and ties the keys of `tagColorMap` to `TagColor` more cleanly

File: packages/gamut-labs/src/InfoCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  topText,
+  title,
+  subtitle,
+  body,
+  bottomLeftText,
+  bottomRightTagText,
+  bottomRightTagColor = 'pink',
+  cardHeight = 285,
+}) => {
+  const renderTopText = (text: string) => (
```
Comment (MattSchiller) [inline packages/gamut-labs/src/InfoCard/index.tsx:39]: Why not pull these functions into small components (even in this file)? It'll save on memory (so each instance of `InfoCard` doesn't have to re-declare them) and it'll make the component overall more legible IMO

**Review by MattSchiller — APPROVED** [review]
Looks good. Only left some code style/type stuff. Niccccce!

File: packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx (hunk 1 above)
Comment (MattSchiller) [reply packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx]: Says outdated, but it is not outdated -_-

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.d3059e.0<br/>@codecademy/gamut-labs@28.5.2-alpha.d3059e.0<br/>@codecademy/gamut@42.2.2-alpha.d3059e.0<br/>@codecademy/monaco@0.5.28-alpha.d3059e.0<br/>@codecademy/styleguide@52.3.2-alpha.d3059e.0<br/></pre>

File: packages/gamut/src/InfoCard/index.tsx (hunk 2 above)
Comment (dreamwasp) [reply packages/gamut/src/InfoCard/index.tsx]: we could also pick these from our theme colors to stop this from breaking in case those tokens ever change

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61c0e05a14dc844bb2d0e19a--gamut-preview.netlify.app>https://61c0e05a14dc844bb2d0e19a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61c0e05a14dc844bb2d0e19a>Deploy Logs</a></p>

File: packages/gamut/src/InfoCard/index.tsx (hunk 2 above)
Comment (MattSchiller) [reply packages/gamut/src/InfoCard/index.tsx]: You mean up in the def for `TagColor`, yea?

Like:
```ts
export type TagColor = Pick<GamutColors, 'blue' | 'green' | 'pink'>
``` ?

File: packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx (hunk 1 above)
Comment (sooinc) [reply packages/gamut/src/InfoCard/__tests__/InfoCard-test.tsx]: lol it's cause i moved it to gamut-labs. but yes good call, addressing this now!

File: packages/gamut/src/InfoCard/index.tsx (hunk 2 above)
Comment (dreamwasp) [reply packages/gamut/src/InfoCard/index.tsx]: yeah!

Comment (sooinc) [reply packages/gamut/src/InfoCard/index.tsx]: Ah sorry just seeing the latest comments. I ended up doing this:
```
export type TagColor = 'blue' | 'green' | 'pink';
const tagColorMap: Record<TagColor, keyof Theme['colors']> = {
  blue: 'paleBlue',
  green: 'paleGreen',
  pink: 'palePink',
};
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.cf23da.0<br/>@codecademy/gamut-labs@28.5.2-alpha.cf23da.0<br/>@codecademy/gamut@42.2.2-alpha.cf23da.0<br/>@codecademy/monaco@0.5.28-alpha.cf23da.0<br/>@codecademy/styleguide@52.3.2-alpha.cf23da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61c0ecfa22ba4d5192a624b9--gamut-preview.netlify.app>https://61c0ecfa22ba4d5192a624b9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61c0ecfa22ba4d5192a624b9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.0c4e38.0<br/>@codecademy/gamut-labs@28.5.2-alpha.0c4e38.0<br/>@codecademy/styleguide@52.3.2-alpha.0c4e38.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61c0eecfbcbea35bab1d33e3--gamut-preview.netlify.app>https://61c0eecfbcbea35bab1d33e3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61c0eecfbcbea35bab1d33e3>Deploy Logs</a></p>

**Pull request merged**